lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -51,11 +51,11 @@ public function __construct($type, array $parts = [])
     /**
      * Adds multiple parts to composite expression.
      *
-     * @param self[]|string[] $parts
+     * @param array<int, self>|array<int, string> $parts
```

Comment:
Should it be `array<int, self|string>`? Unlike parameters in query builder, every expression part is independent and can be either a string or expression.